libs/portal-integration-angular/src/lib/services/portal-dialog.service.ts

```diff
@@ -1,8 +1,424 @@
+   *
+   * @param title Dialog title
+   * @param componentOrMessage Dialog content to display
+   * @param primaryButtonTranslationKeyOrDetails Primary button to display in the bottom of the dialog
+   * @param secondaryButtonTranslationKeyOrDetails Secondary button to display in the bottom of the dialog
+   * @param showXButton Determines if X (close) button should be visible in the top right corner of the dialog (not displayed if set to false and displayed if set to true). NOTE: If only one button is being used then close button will not be visible on the dialog
+   * @returns Observable containing dialog state on close
+   *
+   * Displaying component inisde the dialog can be achieved by providing the component class with optional inputs. By default the component will be shown without any interaction with the dialog, however you can implement the following interfaces by your component class to allow for some interactions:
+   * - {@link DialogResult} - dialog state will contain dialogResult property
+   *
+   * - {@link DialogButtonClicked} - on button click ocxDialogButtonClicked function will be called with dialog state as a parameter. You should return true if you want dialog to be close or false if not and add any operations on your component.
+   *
+   * - {@link DialogPrimaryButtonDisabled} - dialog will use the EventEmitter to determine if the primary button should be disabled
+   *
+   * - {@link DialogSecondaryButtonDisabled} - dialog will use the EventEmitter to determine if the secondary button should be disabled
+   *
+   * @example
+   * Display dialog with message and two buttons using translation keys
+   *
+   * ```
+   * this.portalDialogService.openDialog('TITLE_KEY', 'WELCOME_MESSAGE', 'OK_BUTTON', 'REFRESH_BUTTON')
+   * ```
+   *
+   * @example
+   * Display dialog message with icon and single button
+   *
+   * ```
+   * // Welcome message with question mark icon
+   * const dialogMessage = {
+   *   key: 'WELCOME_MESSAGE',
+   *   icon: 'pi pi-question'
+   * }
+   * this.portalDialogService.openDialog('TITLE_KEY', dialogMessage, 'OK_BUTTON', 'REFRESH_BUTTON')
+   * ```
+   *
+   * @example
+   * Display dialog message with two customized buttons
+   *
+   * ```
+   * // Ok button with check icon
+   * const primaryButton = {
+   *   key: 'OK_BUTTON',
+   *   icon: 'pi pi-check'
+   * }
+   *
+   * // Refresh button with refresh icon
+   * const secondaryButton = {
+   *   key: 'REFRESH_BUTTON',
+   *   icon: 'pi pi-refresh'
+   * }
+   *
+   * this.portalDialogService.openDialog('TITLE_KEY', 'WELCOME_MESSAGE', primaryButton, secondaryButton)
+   * ```
+   *
+   * @example
+   * Display dialog message without X button in top right corner
+   *
+   * ```
+   * this.portalDialogService.openDialog('TITLE_KEY', 'WELCOME_MESSAGE', 'OK_BUTTON', 'REFRESH_BUTTON', false)
+   * ```
+   *
+   * @example
+   * React on dialog closing
```

Comment:
You should mention somewhere in the description of the function (text above) that the dialog is only shown when you are subscribing to the function